web/src/js/__tests__/ducks/_tbackendstate.ts

```diff
@@ -0,0 +1,60 @@
+/** Auto-generated by test_app.py:test_generate_state_js */
+import {BackendState} from '../../ducks/backendState';
+export function TBackendState(): Required<BackendState> {
+    return {
+        "available": true,
+        "contentViews": [
+            "Auto",
+            "Raw",
+            "Hex",
+            "GraphQL",
+            "JSON",
+            "XML/HTML",
+            "WBXML",
+            "JavaScript",
+            "CSS",
+            "URL-encoded",
+            "Multipart Form",
+            "Image",
+            "Protocol Buffer",
+            "MsgPack",
+            "gRPC/Protocol Buffer"
+        ],
+        "servers": [
+            {
+                "description": "HTTP(S) proxy",
+                "full_spec": "regular",
```

Comment:
Looking at this I wonder if we should add a property like `expanded_full_spec` (or call the current just `spec`) that's basically the _actual_ full spec, with all the defaults options filled in. Or maybe also a `human_spec` that `to_json` gives us to show users. Right now `full_spec` isn't the full spec, it's just what the option was set to. I would like both please :smile: . I'll probably just render a list of `<li>{actual_full_spec}</li>` in my UI.

1. Parse spec
2. It sets all the default options not specified in the option
3. Serialize it back with the full options (host, port, etc.) to a spec that would result in the exact same listener but verbose and human friendly